waiter/src/waiter/marathon_api.clj

```diff
@@ -0,0 +1,147 @@
+;;
+;;       Copyright (c) 2017 Two Sigma Investments, LP.
+;;       All Rights Reserved
+;;
+;;       THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF
+;;       Two Sigma Investments, LP.
+;;
+;;       The copyright notice above does not evidence any
+;;       actual or intended publication of such source code.
+;;
+(ns waiter.marathon-api
+  (:require [clojure.core.async :as async]
+            [clojure.data.json :as json]
+            [clojure.walk :as walk]
+            [qbits.jet.client.http :as http]
+            [slingshot.slingshot :as ss]
+            [waiter.auth.spnego :as spnego])
+  (:import (java.net URI)
+           (org.eclipse.jetty.client HttpClient)))
+
+(defprotocol MarathonApi
```

Comment:
What does having the protocol and impl buy us over just having the functions live directly in the namespace?

Reply:
Since we need to inject in an instance, we can't use the statically defined functions. I used the protocol and its implementation to mimic this behavior of objects encapsulating the http-client and spnego config.

Reply:
What do you mean by "inject in an instance"?

Reply:
Being able to wrap these values `[^HttpClient http-client ^String marathon-url spnego-auth]` as a `MarathonApi ` object. Do you prefer I pass these around as a map and not use the `MarathonApi ` protocol?

Reply:
Actually, I liked wrapping them in a map and avoiding the `defprotocol` :)